src/main/solution_space/serialized_code.py

```diff
@@ -102,18 +129,21 @@ def language(self) -> consts.LANGUAGE:
     def is_full(self) -> bool:
         return self._rate == consts.TEST_RESULT.FULL_SOLUTION.value
 
-    def add_anon_tree(self, anon_tree) -> Optional[str]:
-        if self.does_contain_anon_tree(anon_tree):
+    def add_anon_tree(self, anon_tree: ast.AST, code_info: CodeInfo) -> Optional[str]:
+        found_anon_tree = self.find_anon_tree(anon_tree)
+        if found_anon_tree:
+            found_anon_tree.add_code_info(code_info)
             return None
 
-        self._anon_trees.append(anon_tree)
-        index_of_tree = self._anon_trees.index(anon_tree)
-        return self.__create_file_for_tree(anon_tree, f'{TREE_TYPE.ANON.value}_{index_of_tree}')
+        new_anon_tree = AnonTree(anon_tree, code_info)
+        self._anon_trees.append(new_anon_tree)
+        index_of_tree = self._anon_trees.index(new_anon_tree)
```

Comment:
We use simple list and the append function always add items into the end of the list. Maybe should we replace `self._anon_trees.index(new_anon_tree)` to `len(self._anon_trees)`? It would be faster than your implementation